storage: TestLargeUnsplittableRangeReplicate failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/5a0ae02c62d8a1eb50e930eeda634d32be297d58

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLargeUnsplittableRangeReplicate PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1370058&tab=buildLog

```
I190702 07:00:49.446301 654059 util/stop/stopper.go:542  quiescing; tasks left:
1      node.Node: writing summary
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190702 07:00:49.446430 654057 util/stop/stopper.go:542  quiescing; tasks left:
1      [async] closedts-subscription
W190702 07:00:49.446603 642140 gossip/gossip.go:1498  [n2] no incoming or outgoing connections
W190702 07:00:49.446706 643350 storage/raft_transport.go:578  [n2] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
W190702 07:00:49.446908 646698 storage/raft_transport.go:578  [n4] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W190702 07:00:49.447272 643235 storage/raft_transport.go:578  [n4] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
I190702 07:00:49.448430 654060 util/stop/stopper.go:542  quiescing; tasks left:
2      [async] closedts-subscription
I190702 07:00:49.449166 654059 util/stop/stopper.go:542  quiescing; tasks left:
1      node.Node: writing summary
I190702 07:00:49.476426 654032 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:44849: context canceled
I190702 07:00:49.477752 654030 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:43971: context canceled
I190702 07:00:49.478303 654033 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:38225: context canceled
W190702 07:00:49.482034 641482 storage/store.go:1535  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I190702 07:00:49.518538 654042 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:44849: context canceled
W190702 07:00:49.518793 642902 storage/store.go:1535  [n4,s4,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I190702 07:00:49.519608 654045 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:42547: context canceled
I190702 07:00:49.519895 654043 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:41137: context canceled
I190702 07:00:49.528488 654044 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:38225: context canceled
W190702 07:00:49.529725 642582 server/node.go:785  [n3,summaries] error recording status summaries: node unavailable; try another peer
W190702 07:00:49.564710 642902 storage/store.go:1535  [n4,s4,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I190702 07:00:49.574707 654010 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:43971: context canceled
--- FAIL: TestLargeUnsplittableRangeReplicate (72.43s)
    replicate_queue_test.go:390: condition failed to evaluate within 45s: not up-replicated yet
        goroutine 641297 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc002dd8de0, 0x4b3b6e0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4c13f00, 0xc00f8ce000, 0xc0015e45a0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:45 +0x181
        github.com/cockroachdb/cockroach/pkg/storage_test.TestLargeUnsplittableRangeReplicate(0xc00f8ce000)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replicate_queue_test.go:390 +0xbc3
        testing.tRunner(0xc00f8ce000, 0x3fda248)
        	/usr/local/go/src/testing/testing.go:865 +0x164
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:916 +0x65b

```

SHA: https://github.com/cockroachdb/cockroach/commits/9e55b99314f751a2e122e0decfc0160a4dee116f

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLargeUnsplittableRangeReplicate PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1372233&tab=buildLog

```
I190703 07:13:55.934923 707226 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:40093: context canceled
I190703 07:13:55.935665 707274 util/stop/stopper.go:542  quiescing; tasks left:
1      [async] closedts-subscription
I190703 07:13:55.936340 707227 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:38223: context canceled
I190703 07:13:55.949275 696599 storage/queue.go:518  [n3,s3] rate limited in MaybeAdd (gc): node unavailable; try another peer
I190703 07:13:55.949680 696599 storage/queue.go:518  [n3,s3] rate limited in MaybeAdd (merge): node unavailable; try another peer
I190703 07:13:55.949827 696599 storage/queue.go:518  [n3,s3] rate limited in MaybeAdd (split): node unavailable; try another peer
I190703 07:13:55.950100 696599 storage/queue.go:518  [n3,s3] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I190703 07:13:55.950387 696599 storage/queue.go:518  [n3,s3] rate limited in MaybeAdd (replicaGC): node unavailable; try another peer
I190703 07:13:55.950529 696599 storage/queue.go:518  [n3,s3] rate limited in MaybeAdd (raftlog): node unavailable; try another peer
I190703 07:13:55.950741 696599 storage/queue.go:518  [n3,s3] rate limited in MaybeAdd (raftsnapshot): node unavailable; try another peer
I190703 07:13:55.950977 696599 storage/queue.go:518  [n3,s3] rate limited in MaybeAdd (consistencyChecker): node unavailable; try another peer
I190703 07:13:55.951338 696599 storage/queue.go:518  [n3,s3] rate limited in MaybeAdd (timeSeriesMaintenance): node unavailable; try another peer
I190703 07:13:55.957091 707285 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:38223: context canceled
I190703 07:13:55.957149 707286 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:40365: context canceled
I190703 07:13:55.957779 707284 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:45235: context canceled
E190703 07:13:55.957524 707177 vendor/google.golang.org/grpc/pickfirst.go:59  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
I190703 07:13:55.957842 707287 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:39121: context canceled
I190703 07:13:55.973002 707155 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:45235: context canceled
I190703 07:13:55.974575 707152 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:39121: context canceled
I190703 07:13:55.975838 707153 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:40365: context canceled
I190703 07:13:55.976194 707154 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:40093: context canceled
--- FAIL: TestLargeUnsplittableRangeReplicate (67.21s)
    replicate_queue_test.go:390: condition failed to evaluate within 45s: not up-replicated yet
        goroutine 695245 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc00775d800, 0x4e5ba60)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4f34860, 0xc01b690900, 0xc0003edd10)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:45 +0x181
        github.com/cockroachdb/cockroach/pkg/storage_test.TestLargeUnsplittableRangeReplicate(0xc01b690900)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replicate_queue_test.go:390 +0xbc3
        testing.tRunner(0xc01b690900, 0x42f8708)
        	/usr/local/go/src/testing/testing.go:865 +0x164
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:916 +0x65b

```

SHA: https://github.com/cockroachdb/cockroach/commits/4878cb3e960dc26f0946e527e1836f27b3304c97

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLargeUnsplittableRangeReplicate PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1374336&tab=buildLog

```
W190704 11:11:36.025597 689247 storage/raft_transport.go:578  [n5] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = context canceled:
I190704 11:11:36.025739 697970 util/stop/stopper.go:542  quiescing; tasks left:
3      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190704 11:11:36.030979 698008 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:39379: context canceled
I190704 11:11:36.030087 697971 util/stop/stopper.go:542  quiescing; tasks left:
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190704 11:11:36.029944 697967 util/stop/stopper.go:542  quiescing; tasks left:
1      [async] transport racer
I190704 11:11:36.031808 698009 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:42097: context canceled
I190704 11:11:36.030779 698007 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:34251: context canceled
I190704 11:11:36.041523 698101 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:39051: context canceled
I190704 11:11:36.040542 686792 storage/queue.go:518  [n1,s1] rate limited in MaybeAdd (gc): node unavailable; try another peer
I190704 11:11:36.044537 686792 storage/queue.go:518  [n1,s1] rate limited in MaybeAdd (merge): node unavailable; try another peer
I190704 11:11:36.044717 698103 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:34251: context canceled
I190704 11:11:36.044801 686792 storage/queue.go:518  [n1,s1] rate limited in MaybeAdd (split): node unavailable; try another peer
I190704 11:11:36.047445 686792 storage/queue.go:518  [n1,s1] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I190704 11:11:36.047686 686792 storage/queue.go:518  [n1,s1] rate limited in MaybeAdd (replicaGC): node unavailable; try another peer
I190704 11:11:36.047853 686792 storage/queue.go:518  [n1,s1] rate limited in MaybeAdd (raftlog): node unavailable; try another peer
I190704 11:11:36.048038 686792 storage/queue.go:518  [n1,s1] rate limited in MaybeAdd (raftsnapshot): node unavailable; try another peer
I190704 11:11:36.048162 686792 storage/queue.go:518  [n1,s1] rate limited in MaybeAdd (consistencyChecker): node unavailable; try another peer
I190704 11:11:36.048316 686792 storage/queue.go:518  [n1,s1] rate limited in MaybeAdd (timeSeriesMaintenance): node unavailable; try another peer
I190704 11:11:36.044859 698102 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:42097: context canceled
I190704 11:11:36.041065 697972 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:39379: context canceled
I190704 11:11:36.088694 686817 kv/transport_race.go:108  transport race promotion: ran 69 iterations on up to 4034 requests
--- FAIL: TestLargeUnsplittableRangeReplicate (66.83s)
    replicate_queue_test.go:390: condition failed to evaluate within 45s: not up-replicated yet
        goroutine 686615 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc013a8b6b0, 0x4e5d760)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4f36660, 0xc008273500, 0xc008334990)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:45 +0x181
        github.com/cockroachdb/cockroach/pkg/storage_test.TestLargeUnsplittableRangeReplicate(0xc008273500)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replicate_queue_test.go:390 +0xbc3
        testing.tRunner(0xc008273500, 0x42fa210)
        	/usr/local/go/src/testing/testing.go:865 +0x164
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:916 +0x65b

```

SHA: https://github.com/cockroachdb/cockroach/commits/a2f104c467ee1553a875b45f1e16f8cfb3e3195d

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLargeUnsplittableRangeReplicate PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1375103&tab=buildLog

```
I190705 10:35:18.875188 637183 util/stop/stopper.go:542  quiescing; tasks left:
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190705 10:35:18.879225 637183 util/stop/stopper.go:542  quiescing; tasks left:
1      [async] closedts-subscription
I190705 10:35:18.879413 637185 util/stop/stopper.go:542  quiescing; tasks left:
3      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190705 10:35:18.879896 637185 util/stop/stopper.go:542  quiescing; tasks left:
2      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190705 10:35:18.881612 637185 util/stop/stopper.go:542  quiescing; tasks left:
2      [async] closedts-subscription
I190705 10:35:18.882053 637185 util/stop/stopper.go:542  quiescing; tasks left:
1      [async] closedts-subscription
I190705 10:35:18.883881 637187 util/stop/stopper.go:542  quiescing; tasks left:
3      [async] closedts-subscription
I190705 10:35:18.885219 637187 util/stop/stopper.go:542  quiescing; tasks left:
2      [async] closedts-subscription
I190705 10:35:18.893332 637270 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:41711: context canceled
I190705 10:35:18.895074 637268 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:41851: context canceled
I190705 10:35:18.895520 637269 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:39653: context canceled
I190705 10:35:18.896362 637261 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:35049: context canceled
I190705 10:35:18.896820 637271 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:32975: context canceled
I190705 10:35:18.897816 637263 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:39653: context canceled
I190705 10:35:18.898447 637262 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:41711: context canceled
E190705 10:35:18.901587 637169 vendor/google.golang.org/grpc/pickfirst.go:59  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
I190705 10:35:18.901879 637264 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:41851: context canceled
I190705 10:35:18.984329 637243 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:35049: context canceled
I190705 10:35:18.984499 637244 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:39653: context canceled
I190705 10:35:18.984622 637245 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:32975: context canceled
I190705 10:35:18.986287 637242 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:41851: context canceled
--- FAIL: TestLargeUnsplittableRangeReplicate (68.12s)
    replicate_queue_test.go:390: condition failed to evaluate within 45s: not up-replicated yet
        goroutine 625464 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc011df9a10, 0x4e6ba40)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4f44f60, 0xc013eba200, 0xc0068b0440)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:45 +0x181
        github.com/cockroachdb/cockroach/pkg/storage_test.TestLargeUnsplittableRangeReplicate(0xc013eba200)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replicate_queue_test.go:390 +0xbc3
        testing.tRunner(0xc013eba200, 0x4306a78)
        	/usr/local/go/src/testing/testing.go:865 +0x164
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:916 +0x65b

```

SHA: https://github.com/cockroachdb/cockroach/commits/9322e07476de447799c5d3011eb2874930ee2993

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLargeUnsplittableRangeReplicate PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1375922&tab=buildLog

```
W190706 10:52:04.095567 676392 gossip/gossip.go:1498  [n3] no incoming or outgoing connections
W190706 10:52:04.096566 676685 storage/raft_transport.go:578  [n3] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
I190706 10:52:04.097379 687405 util/stop/stopper.go:542  quiescing; tasks left:
1      [async] closedts-subscription
W190706 10:52:04.097915 676465 storage/raft_transport.go:578  [n1] while processing outgoing Raft queue to node 3: EOF:
W190706 10:52:04.100430 677278 storage/raft_transport.go:578  [n4] while processing outgoing Raft queue to node 1: rpc error: code = Unavailable desc = transport is closing:
I190706 10:52:04.102640 687418 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:33007: context canceled
I190706 10:52:04.102855 687417 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:33033: context canceled
I190706 10:52:04.103700 687404 util/stop/stopper.go:542  quiescing; tasks left:
4      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190706 10:52:04.104495 687419 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:42485: context canceled
I190706 10:52:04.106659 687420 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:38043: context canceled
I190706 10:52:04.107075 687403 util/stop/stopper.go:542  quiescing; tasks left:
3      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
W190706 10:52:04.108457 676993 storage/raft_transport.go:578  [n1] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
I190706 10:52:04.108797 687403 util/stop/stopper.go:542  quiescing; tasks left:
2      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190706 10:52:04.109566 687406 util/stop/stopper.go:542  quiescing; tasks left:
2      [async] closedts-subscription
I190706 10:52:04.109662 687403 util/stop/stopper.go:542  quiescing; tasks left:
1      [async] closedts-subscription
I190706 10:52:04.110233 687404 util/stop/stopper.go:542  quiescing; tasks left:
2      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190706 10:52:04.110714 687404 util/stop/stopper.go:542  quiescing; tasks left:
1      [async] closedts-subscription
I190706 10:52:04.130420 687449 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:38043: context canceled
I190706 10:52:04.130608 687450 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:33683: context canceled
I190706 10:52:04.131017 687451 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:33033: context canceled
I190706 10:52:04.130832 687452 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:33007: context canceled
--- FAIL: TestLargeUnsplittableRangeReplicate (68.13s)
    replicate_queue_test.go:390: condition failed to evaluate within 45s: not up-replicated yet
        goroutine 675170 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc006518bd0, 0x4e6ba40)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4f44f60, 0xc00e4a4a00, 0xc017676290)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:45 +0x181
        github.com/cockroachdb/cockroach/pkg/storage_test.TestLargeUnsplittableRangeReplicate(0xc00e4a4a00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replicate_queue_test.go:390 +0xbc3
        testing.tRunner(0xc00e4a4a00, 0x4306a78)
        	/usr/local/go/src/testing/testing.go:865 +0x164
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:916 +0x65b

```

SHA: https://github.com/cockroachdb/cockroach/commits/36b74f0373b040aa873d4c23b415d94deba88378

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLargeUnsplittableRangeReplicate PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1376818&tab=buildLog

```
W190708 05:56:20.418880 721507 internal/client/txn.go:506  [n1,replicate,s1,r30/1:/{Table/52/1/2-Max}] failure aborting transaction: node unavailable; try another peer; abort caused by: failed to send RPC: sending to all 5 replicas failed; last error: <nil> failed to connect to n3 at 127.0.0.1:37535: context canceled
E190708 05:56:20.419494 721507 storage/queue.go:1027  [n1,replicate,s1,r30/1:/{Table/52/1/2-Max}] change replicas of r30 failed: failed to send RPC: sending to all 5 replicas failed; last error: <nil> failed to connect to n3 at 127.0.0.1:37535: context canceled
W190708 05:56:20.420182 714320 storage/raft_transport.go:578  [n4] while processing outgoing Raft queue to node 5: EOF:
I190708 05:56:20.421061 721755 util/stop/stopper.go:542  quiescing; tasks left:
4      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
W190708 05:56:20.423063 712709 storage/raft_transport.go:578  [n3] while processing outgoing Raft queue to node 5: EOF:
W190708 05:56:20.437061 714337 storage/raft_transport.go:578  [n5] while processing outgoing Raft queue to node 4: rpc error: code = Canceled desc = context canceled:
W190708 05:56:20.439452 712044 storage/raft_transport.go:578  [n3] while processing outgoing Raft queue to node 4: rpc error: code = Canceled desc = grpc: the client connection is closing:
I190708 05:56:20.440110 721756 util/stop/stopper.go:542  quiescing; tasks left:
4      [async] closedts-subscription
1      [async] intent_resolver_ir_batcher
1      [async] closedts-rangefeed-subscriber
I190708 05:56:20.441463 721756 util/stop/stopper.go:542  quiescing; tasks left:
4      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190708 05:56:20.442828 721756 util/stop/stopper.go:542  quiescing; tasks left:
3      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190708 05:56:20.443047 721754 util/stop/stopper.go:542  quiescing; tasks left:
4      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190708 05:56:20.443471 721756 util/stop/stopper.go:542  quiescing; tasks left:
2      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
W190708 05:56:20.444247 712646 storage/raft_transport.go:578  [n5] while processing outgoing Raft queue to node 3: EOF:
I190708 05:56:20.444350 721756 util/stop/stopper.go:542  quiescing; tasks left:
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
W190708 05:56:20.445073 712065 storage/raft_transport.go:578  [n4] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = context canceled:
I190708 05:56:20.447903 721755 util/stop/stopper.go:542  quiescing; tasks left:
2      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190708 05:56:20.448314 721754 util/stop/stopper.go:542  quiescing; tasks left:
3      [async] closedts-subscription
I190708 05:56:20.448807 721754 util/stop/stopper.go:542  quiescing; tasks left:
2      [async] closedts-subscription
I190708 05:56:20.449256 721754 util/stop/stopper.go:542  quiescing; tasks left:
1      [async] closedts-subscription
W190708 05:56:20.499464 711525 gossip/gossip.go:1498  [n4] no incoming or outgoing connections
--- FAIL: TestLargeUnsplittableRangeReplicate (65.80s)
    replicate_queue_test.go:390: condition failed to evaluate within 45s: not up-replicated yet
        goroutine 710046 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc0095ccc00, 0x4e6ba40)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4f44f60, 0xc013579a00, 0xc0133c0740)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:45 +0x181
        github.com/cockroachdb/cockroach/pkg/storage_test.TestLargeUnsplittableRangeReplicate(0xc013579a00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replicate_queue_test.go:390 +0xbc3
        testing.tRunner(0xc013579a00, 0x4306a78)
        	/usr/local/go/src/testing/testing.go:865 +0x164
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:916 +0x65b

```

SHA: https://github.com/cockroachdb/cockroach/commits/813b0146e763e8f44c4cb45e498e68e6ace9bc2d

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLargeUnsplittableRangeReplicate PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1381594&tab=buildLog

```
I190710 10:29:36.899783 618389 storage/queue.go:518  [n3,s3] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I190710 10:29:36.899907 618389 storage/queue.go:518  [n3,s3] rate limited in MaybeAdd (replicaGC): node unavailable; try another peer
I190710 10:29:36.900051 618389 storage/queue.go:518  [n3,s3] rate limited in MaybeAdd (raftlog): node unavailable; try another peer
I190710 10:29:36.900222 618389 storage/queue.go:518  [n3,s3] rate limited in MaybeAdd (raftsnapshot): node unavailable; try another peer
I190710 10:29:36.900342 618389 storage/queue.go:518  [n3,s3] rate limited in MaybeAdd (consistencyChecker): node unavailable; try another peer
I190710 10:29:36.900505 618389 storage/queue.go:518  [n3,s3] rate limited in MaybeAdd (timeSeriesMaintenance): node unavailable; try another peer
I190710 10:29:36.927634 628669 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:35057: context canceled
I190710 10:29:36.927726 628670 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:43151: context canceled
I190710 10:29:36.928016 628671 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:39647: context canceled
I190710 10:29:36.928690 628672 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:40921: context canceled
I190710 10:29:36.982119 628625 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:43207: context canceled
I190710 10:29:36.982686 628626 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:43151: context canceled
I190710 10:29:36.983270 628624 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:35057: context canceled
I190710 10:29:36.984054 628623 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:39647: context canceled
I190710 10:29:36.993948 628676 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:40921: context canceled
I190710 10:29:36.995380 628673 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:43207: context canceled
I190710 10:29:36.996560 628674 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:43151: context canceled
E190710 10:29:37.128116 628699 vendor/google.golang.org/grpc/pickfirst.go:59  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
E190710 10:29:37.128229 628690 vendor/google.golang.org/grpc/pickfirst.go:59  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
I190710 10:29:37.134732 617044 kv/transport_race.go:108  transport race promotion: ran 112 iterations on up to 4003 requests
--- FAIL: TestLargeUnsplittableRangeReplicate (67.67s)
    soon.go:35: condition failed to evaluate within 45s: not up-replicated yet. replicas: {1,2,3}
        goroutine 617102 [running]:
        runtime/debug.Stack(0x525ba40, 0xc004aaa900, 0xc01132c1b0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x525ba40, 0xc004aaa900, 0xc01132c1b0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/storage_test.TestLargeUnsplittableRangeReplicate(0xc004aaa900)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replicate_queue_test.go:390 +0xbc3
        testing.tRunner(0xc004aaa900, 0x45d8e28)
        	/usr/local/go/src/testing/testing.go:865 +0x164
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:916 +0x65b

```

SHA: https://github.com/cockroachdb/cockroach/commits/e41c2daee9dffc3a19d59cd176d97e8086b569e2

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLargeUnsplittableRangeReplicate PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1383659&tab=buildLog

```
1      [async] storage.timeSeriesMaintenance: processing replica
I190711 07:25:56.061673 642495 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:45679: context canceled
I190711 07:25:56.064819 642493 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:40269: context canceled
I190711 07:25:56.095514 642504 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:40269: context canceled
I190711 07:25:56.096967 642505 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:45679: context canceled
W190711 07:25:56.097516 642439 storage/idalloc/id_alloc.go:101  [n3,s3] node unavailable; try another peer
E190711 07:25:56.098043 642438 storage/queue.go:1027  [n3,split,s3,r24/3:/System/tsd/cr.store.{qu…-ra…}] unable to allocate right hand side range descriptor: could not allocate ID; system is draining
I190711 07:25:56.121538 632520 rpc/nodedialer/nodedialer.go:139  [n3] unable to connect to n5: failed to connect to n5 at 127.0.0.1:45679: context canceled
E190711 07:25:56.187690 642514 vendor/google.golang.org/grpc/pickfirst.go:59  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
I190711 07:25:56.218314 632433 storage/queue.go:518  [n3,s3] rate limited in MaybeAdd (gc): node unavailable; try another peer
I190711 07:25:56.218760 632433 storage/queue.go:518  [n3,s3] rate limited in MaybeAdd (merge): node unavailable; try another peer
I190711 07:25:56.218914 632433 storage/queue.go:518  [n3,s3] rate limited in MaybeAdd (split): node unavailable; try another peer
I190711 07:25:56.219077 632433 storage/queue.go:518  [n3,s3] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I190711 07:25:56.219217 632433 storage/queue.go:518  [n3,s3] rate limited in MaybeAdd (replicaGC): node unavailable; try another peer
I190711 07:25:56.219326 632433 storage/queue.go:518  [n3,s3] rate limited in MaybeAdd (raftlog): node unavailable; try another peer
I190711 07:25:56.219442 632433 storage/queue.go:518  [n3,s3] rate limited in MaybeAdd (raftsnapshot): node unavailable; try another peer
I190711 07:25:56.219585 632433 storage/queue.go:518  [n3,s3] rate limited in MaybeAdd (consistencyChecker): node unavailable; try another peer
I190711 07:25:56.219750 632433 storage/queue.go:518  [n3,s3] rate limited in MaybeAdd (timeSeriesMaintenance): node unavailable; try another peer
I190711 07:25:56.222377 642539 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:33945: context canceled
W190711 07:25:56.234525 631425 storage/store.go:1535  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W190711 07:25:56.282993 631425 storage/store.go:1535  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I190711 07:25:56.384548 631437 kv/transport_race.go:108  transport race promotion: ran 61 iterations on up to 4100 requests
--- FAIL: TestLargeUnsplittableRangeReplicate (65.70s)
    soon.go:35: condition failed to evaluate within 45s: not up-replicated yet. replicas: {1,2,3}
        goroutine 631303 [running]:
        runtime/debug.Stack(0x50bc060, 0xc0015c3900, 0xc0038684a0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x50bc060, 0xc0015c3900, 0xc0038684a0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/storage_test.TestLargeUnsplittableRangeReplicate(0xc0015c3900)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replicate_queue_test.go:390 +0xbc3
        testing.tRunner(0xc0015c3900, 0x43e5728)
        	/usr/local/go/src/testing/testing.go:865 +0x164
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:916 +0x65b

```

SHA: https://github.com/cockroachdb/cockroach/commits/35a957115b936d8f2df8f38566630c6930a95088

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLargeUnsplittableRangeReplicate PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1385720&tab=buildLog

```
4      [async] closedts-subscription
I190712 12:42:50.533928 639911 util/stop/stopper.go:542  quiescing; tasks left:
3      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
W190712 12:42:50.535912 629863 gossip/gossip.go:1498  [n4] no incoming or outgoing connections
I190712 12:42:50.538280 639913 util/stop/stopper.go:542  quiescing; tasks left:
2      [async] closedts-subscription
W190712 12:42:50.538854 633350 storage/raft_transport.go:578  [n3] while processing outgoing Raft queue to node 4: rpc error: code = Canceled desc = context canceled:
I190712 12:42:50.539425 639912 util/stop/stopper.go:542  quiescing; tasks left:
3      [async] closedts-subscription
I190712 12:42:50.540093 639912 util/stop/stopper.go:542  quiescing; tasks left:
2      [async] closedts-subscription
W190712 12:42:50.542647 633072 storage/raft_transport.go:578  [n3] while processing outgoing Raft queue to node 5: rpc error: code = Canceled desc = context canceled:
I190712 12:42:50.543143 639811 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:43009: context canceled
I190712 12:42:50.543762 639914 util/stop/stopper.go:542  quiescing; tasks left:
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190712 12:42:50.543987 639957 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:40685: context canceled
I190712 12:42:50.544228 639973 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:40685: context canceled
W190712 12:42:50.545210 629512 gossip/gossip.go:1498  [n3] no incoming or outgoing connections
I190712 12:42:50.545399 639911 util/stop/stopper.go:542  quiescing; tasks left:
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190712 12:42:50.545960 639956 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:43123: context canceled
I190712 12:42:50.546513 639843 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:36649: context canceled
I190712 12:42:50.546548 639972 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:46245: context canceled
I190712 12:42:50.547089 639974 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:43123: context canceled
I190712 12:42:50.552725 639937 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:40685: context canceled
I190712 12:42:50.553897 639936 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:46245: context canceled
I190712 12:42:50.555180 639938 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:36649: context canceled
I190712 12:42:50.559987 639939 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:43009: context canceled
--- FAIL: TestLargeUnsplittableRangeReplicate (66.15s)
    soon.go:35: condition failed to evaluate within 45s: not up-replicated yet. replicas: {1,2,3}
        goroutine 628420 [running]:
        runtime/debug.Stack(0x52ba8c0, 0xc008875e00, 0xc003b4c680)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x52ba8c0, 0xc008875e00, 0xc003b4c680)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/storage_test.TestLargeUnsplittableRangeReplicate(0xc008875e00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replicate_queue_test.go:390 +0xbc3
        testing.tRunner(0xc008875e00, 0x455dd90)
        	/usr/local/go/src/testing/testing.go:865 +0x164
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:916 +0x65b

```

SHA: https://github.com/cockroachdb/cockroach/commits/092e60a978e1baec6289a791ee410bc3ec02d1a7

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLargeUnsplittableRangeReplicate PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1387582&tab=buildLog

```
1      [async] closedts-rangefeed-subscriber
I190713 12:40:10.222322 702759 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:33199: context canceled
I190713 12:40:10.223318 702761 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:42449: context canceled
I190713 12:40:10.223669 702762 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:39615: context canceled
I190713 12:40:10.224641 702760 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:37315: context canceled
I190713 12:40:10.235295 691570 kv/transport_race.go:108  transport race promotion: ran 147 iterations on up to 3819 requests
I190713 12:40:10.237735 693299 storage/queue.go:518  [n4,s4] rate limited in MaybeAdd (gc): node unavailable; try another peer
I190713 12:40:10.238111 693299 storage/queue.go:518  [n4,s4] rate limited in MaybeAdd (merge): node unavailable; try another peer
I190713 12:40:10.238481 693299 storage/queue.go:518  [n4,s4] rate limited in MaybeAdd (split): node unavailable; try another peer
I190713 12:40:10.238778 693299 storage/queue.go:518  [n4,s4] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I190713 12:40:10.238978 693299 storage/queue.go:518  [n4,s4] rate limited in MaybeAdd (replicaGC): node unavailable; try another peer
I190713 12:40:10.239203 693299 storage/queue.go:518  [n4,s4] rate limited in MaybeAdd (raftlog): node unavailable; try another peer
I190713 12:40:10.239412 693299 storage/queue.go:518  [n4,s4] rate limited in MaybeAdd (raftsnapshot): node unavailable; try another peer
I190713 12:40:10.239642 693299 storage/queue.go:518  [n4,s4] rate limited in MaybeAdd (consistencyChecker): node unavailable; try another peer
I190713 12:40:10.239849 692478 storage/queue.go:518  [n2,s2] rate limited in MaybeAdd (gc): node unavailable; try another peer
I190713 12:40:10.240353 692478 storage/queue.go:518  [n2,s2] rate limited in MaybeAdd (merge): node unavailable; try another peer
I190713 12:40:10.240613 692478 storage/queue.go:518  [n2,s2] rate limited in MaybeAdd (split): node unavailable; try another peer
I190713 12:40:10.240746 692478 storage/queue.go:518  [n2,s2] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I190713 12:40:10.240876 692478 storage/queue.go:518  [n2,s2] rate limited in MaybeAdd (replicaGC): node unavailable; try another peer
I190713 12:40:10.241054 692478 storage/queue.go:518  [n2,s2] rate limited in MaybeAdd (raftlog): node unavailable; try another peer
I190713 12:40:10.241212 692478 storage/queue.go:518  [n2,s2] rate limited in MaybeAdd (raftsnapshot): node unavailable; try another peer
I190713 12:40:10.241414 692478 storage/queue.go:518  [n2,s2] rate limited in MaybeAdd (consistencyChecker): node unavailable; try another peer
I190713 12:40:10.241586 692478 storage/queue.go:518  [n2,s2] rate limited in MaybeAdd (timeSeriesMaintenance): node unavailable; try another peer
I190713 12:40:10.239906 693299 storage/queue.go:518  [n4,s4] rate limited in MaybeAdd (timeSeriesMaintenance): node unavailable; try another peer
--- FAIL: TestLargeUnsplittableRangeReplicate (65.62s)
    soon.go:35: condition failed to evaluate within 45s: not up-replicated yet. replicas: {1,2,3}
        goroutine 691621 [running]:
        runtime/debug.Stack(0x5546f00, 0xc00c57c400, 0xc00a409fc0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x5546f00, 0xc00c57c400, 0xc00a409fc0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/storage_test.TestLargeUnsplittableRangeReplicate(0xc00c57c400)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replicate_queue_test.go:390 +0xbc3
        testing.tRunner(0xc00c57c400, 0x47a4620)
        	/usr/local/go/src/testing/testing.go:865 +0x164
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:916 +0x65b

```

SHA: https://github.com/cockroachdb/cockroach/commits/e3b454452c5cb251958e2e5d91b1854a478614c9

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLargeUnsplittableRangeReplicate PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1389892&tab=buildLog

```
1      [async] closedts-rangefeed-subscriber
I190716 09:32:19.787860 643021 util/stop/stopper.go:542  quiescing; tasks left:
2      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190716 09:32:19.788342 643021 util/stop/stopper.go:542  quiescing; tasks left:
2      [async] closedts-subscription
I190716 09:32:19.788713 643020 util/stop/stopper.go:542  quiescing; tasks left:
2      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190716 09:32:19.788766 643021 util/stop/stopper.go:542  quiescing; tasks left:
1      [async] closedts-subscription
I190716 09:32:19.789298 643020 util/stop/stopper.go:542  quiescing; tasks left:
1      [async] closedts-subscription
I190716 09:32:19.796789 631623 storage/queue.go:518  [n2,s2] rate limited in MaybeAdd (gc): node unavailable; try another peer
I190716 09:32:19.797230 631623 storage/queue.go:518  [n2,s2] rate limited in MaybeAdd (merge): node unavailable; try another peer
I190716 09:32:19.797480 631623 storage/queue.go:518  [n2,s2] rate limited in MaybeAdd (split): node unavailable; try another peer
I190716 09:32:19.797629 631623 storage/queue.go:518  [n2,s2] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I190716 09:32:19.797753 631623 storage/queue.go:518  [n2,s2] rate limited in MaybeAdd (replicaGC): node unavailable; try another peer
I190716 09:32:19.797893 631623 storage/queue.go:518  [n2,s2] rate limited in MaybeAdd (raftlog): node unavailable; try another peer
I190716 09:32:19.798037 631623 storage/queue.go:518  [n2,s2] rate limited in MaybeAdd (raftsnapshot): node unavailable; try another peer
I190716 09:32:19.798176 631623 storage/queue.go:518  [n2,s2] rate limited in MaybeAdd (consistencyChecker): node unavailable; try another peer
I190716 09:32:19.798310 631623 storage/queue.go:518  [n2,s2] rate limited in MaybeAdd (timeSeriesMaintenance): node unavailable; try another peer
I190716 09:32:19.802523 643032 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:35397: context canceled
I190716 09:32:19.802944 643033 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:35889: context canceled
I190716 09:32:19.803651 643035 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:40405: context canceled
I190716 09:32:19.804676 643034 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:33177: context canceled
I190716 09:32:19.842816 643062 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:43929: context canceled
I190716 09:32:19.844027 642964 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:35397: context canceled
I190716 09:32:19.845298 643061 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:40405: context canceled
I190716 09:32:19.847970 643063 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:35889: context canceled
--- FAIL: TestLargeUnsplittableRangeReplicate (67.04s)
    soon.go:35: condition failed to evaluate within 45s: not up-replicated yet. replicas: {1,2,3,4}
        goroutine 630709 [running]:
        runtime/debug.Stack(0x5546920, 0xc00af0e600, 0xc00027df70)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x5546920, 0xc00af0e600, 0xc00027df70)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/storage_test.TestLargeUnsplittableRangeReplicate(0xc00af0e600)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replicate_queue_test.go:390 +0xbc3
        testing.tRunner(0xc00af0e600, 0x47a5f70)
        	/usr/local/go/src/testing/testing.go:865 +0x164
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:916 +0x65b

```

SHA: https://github.com/cockroachdb/cockroach/commits/90fc142c8b2bab94f5049e8331e409be74947873

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLargeUnsplittableRangeReplicate PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1392477&tab=buildLog

```
I190717 07:39:21.698173 670049 util/stop/stopper.go:542  quiescing; tasks left:
2      [async] closedts-subscription
I190717 07:39:21.699254 658255 storage/queue.go:518  [n1,s1] rate limited in MaybeAdd (consistencyChecker): node unavailable; try another peer
I190717 07:39:21.698219 670050 util/stop/stopper.go:542  quiescing; tasks left:
3      [async] closedts-subscription
I190717 07:39:21.699813 658255 storage/queue.go:518  [n1,s1] rate limited in MaybeAdd (timeSeriesMaintenance): node unavailable; try another peer
I190717 07:39:21.698723 670051 util/stop/stopper.go:542  quiescing; tasks left:
1      [async] closedts-subscription
I190717 07:39:21.718863 670137 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:43865: context canceled
I190717 07:39:21.718915 670138 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:38283: context canceled
I190717 07:39:21.719621 670140 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:45229: context canceled
I190717 07:39:21.719892 670139 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:34635: context canceled
I190717 07:39:21.768319 670168 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:34635: context canceled
I190717 07:39:21.768987 670167 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:45229: context canceled
I190717 07:39:21.769031 670169 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:43865: context canceled
I190717 07:39:21.774779 670093 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:43865: context canceled
E190717 07:39:21.775004 670142 vendor/google.golang.org/grpc/pickfirst.go:59  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
I190717 07:39:21.775275 670094 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:38283: context canceled
I190717 07:39:21.779382 658426 kv/transport_race.go:108  transport race promotion: ran 76 iterations on up to 4235 requests
W190717 07:39:21.784014 659837 storage/store.go:1535  [n4,s4,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I190717 07:39:21.784636 670017 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:34635: context canceled
I190717 07:39:21.785394 670020 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:38283: context canceled
I190717 07:39:21.786606 670018 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:45229: context canceled
W190717 07:39:21.834394 659837 storage/store.go:1535  [n4,s4,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
--- FAIL: TestLargeUnsplittableRangeReplicate (65.72s)
    soon.go:35: condition failed to evaluate within 45s: not up-replicated yet. replicas: {1,2,3}
        goroutine 658219 [running]:
        runtime/debug.Stack(0x5564a60, 0xc01254dc00, 0xc017a67f20)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x5564a60, 0xc01254dc00, 0xc017a67f20)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/storage_test.TestLargeUnsplittableRangeReplicate(0xc01254dc00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replicate_queue_test.go:390 +0xbc3
        testing.tRunner(0xc01254dc00, 0x47bdc90)
        	/usr/local/go/src/testing/testing.go:865 +0x164
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:916 +0x65b

```

SHA: https://github.com/cockroachdb/cockroach/commits/6d644dfb6d38c3ed2e54a548d909950f6b982490

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLargeUnsplittableRangeReplicate PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1394395&tab=buildLog

```
I190718 13:27:32.779750 640970 storage/queue.go:518  [n5,s5] rate limited in MaybeAdd (raftlog): node unavailable; try another peer
I190718 13:27:32.779953 640970 storage/queue.go:518  [n5,s5] rate limited in MaybeAdd (raftsnapshot): node unavailable; try another peer
I190718 13:27:32.780183 640970 storage/queue.go:518  [n5,s5] rate limited in MaybeAdd (consistencyChecker): node unavailable; try another peer
I190718 13:27:32.781014 640970 storage/queue.go:518  [n5,s5] rate limited in MaybeAdd (timeSeriesMaintenance): node unavailable; try another peer
I190718 13:27:32.780709 650677 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:35341: context canceled
I190718 13:27:32.782034 650580 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:36327: context canceled
I190718 13:27:32.782703 650678 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:36591: context canceled
I190718 13:27:32.784541 650579 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:34125: context canceled
W190718 13:27:32.796578 640000 storage/store.go:1535  [n3,s3,r1/4:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I190718 13:27:32.838892 640368 storage/queue.go:518  [n4,s4] rate limited in MaybeAdd (gc): node unavailable; try another peer
I190718 13:27:32.839283 640368 storage/queue.go:518  [n4,s4] rate limited in MaybeAdd (merge): node unavailable; try another peer
I190718 13:27:32.839515 640368 storage/queue.go:518  [n4,s4] rate limited in MaybeAdd (split): node unavailable; try another peer
I190718 13:27:32.839739 640368 storage/queue.go:518  [n4,s4] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I190718 13:27:32.839842 640368 storage/queue.go:518  [n4,s4] rate limited in MaybeAdd (replicaGC): node unavailable; try another peer
I190718 13:27:32.839921 640368 storage/queue.go:518  [n4,s4] rate limited in MaybeAdd (raftlog): node unavailable; try another peer
I190718 13:27:32.839977 640368 storage/queue.go:518  [n4,s4] rate limited in MaybeAdd (raftsnapshot): node unavailable; try another peer
I190718 13:27:32.840055 640368 storage/queue.go:518  [n4,s4] rate limited in MaybeAdd (consistencyChecker): node unavailable; try another peer
I190718 13:27:32.840112 640368 storage/queue.go:518  [n4,s4] rate limited in MaybeAdd (timeSeriesMaintenance): node unavailable; try another peer
I190718 13:27:32.843351 650589 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:35341: context canceled
I190718 13:27:32.843661 650590 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:41255: context canceled
I190718 13:27:32.843880 650591 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:36327: context canceled
I190718 13:27:32.843908 650592 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:34125: context canceled
--- FAIL: TestLargeUnsplittableRangeReplicate (67.25s)
    soon.go:35: condition failed to evaluate within 45s: not up-replicated yet. replicas: {1,2,3}
        goroutine 638761 [running]:
        runtime/debug.Stack(0x55a0f60, 0xc008e2c700, 0xc019084340)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x55a0f60, 0xc008e2c700, 0xc019084340)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/storage_test.TestLargeUnsplittableRangeReplicate(0xc008e2c700)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replicate_queue_test.go:390 +0xbc3
        testing.tRunner(0xc008e2c700, 0x47e8448)
        	/usr/local/go/src/testing/testing.go:865 +0x164
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:916 +0x65b

```

SHA: https://github.com/cockroachdb/cockroach/commits/1ca35fc4a0e2665e7f6efd945e65a0db97984fa7

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLargeUnsplittableRangeReplicate PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1396497&tab=buildLog

```
W190719 09:11:59.710605 663626 storage/raft_transport.go:578  [n3] while processing outgoing Raft queue to node 2: EOF:
I190719 09:11:59.713441 673752 util/stop/stopper.go:542  quiescing; tasks left:
2      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190719 09:11:59.716576 673751 util/stop/stopper.go:542  quiescing; tasks left:
1      [async] closedts-subscription
I190719 09:11:59.718933 673755 util/stop/stopper.go:542  quiescing; tasks left:
2      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
W190719 09:11:59.719368 663017 storage/raft_transport.go:578  [n3] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
I190719 09:11:59.719970 673752 util/stop/stopper.go:542  quiescing; tasks left:
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190719 09:11:59.720106 673755 util/stop/stopper.go:542  quiescing; tasks left:
2      [async] closedts-subscription
I190719 09:11:59.720480 673752 util/stop/stopper.go:542  quiescing; tasks left:
1      [async] closedts-rangefeed-subscriber
I190719 09:11:59.720832 673755 util/stop/stopper.go:542  quiescing; tasks left:
1      [async] closedts-subscription
I190719 09:11:59.736185 662879 rpc/nodedialer/nodedialer.go:139  [n3] unable to connect to n2: failed to connect to n2 at 127.0.0.1:44965: context canceled
I190719 09:11:59.737335 662879 rpc/nodedialer/nodedialer.go:139  [n3] unable to connect to n5: failed to connect to n5 at 127.0.0.1:43167: context canceled
I190719 09:11:59.738203 662879 rpc/nodedialer/nodedialer.go:139  [n3] unable to connect to n1: failed to connect to n1 at 127.0.0.1:43109: context canceled
I190719 09:11:59.739467 662879 rpc/nodedialer/nodedialer.go:139  [n3] unable to connect to n4: failed to connect to n4 at 127.0.0.1:46693: context canceled
I190719 09:11:59.758795 673784 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:40257: context canceled
I190719 09:11:59.759224 673786 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:43167: context canceled
I190719 09:11:59.759465 673787 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:43109: context canceled
I190719 09:11:59.759300 673785 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:46693: context canceled
I190719 09:11:59.796536 673758 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:40257: context canceled
I190719 09:11:59.797177 673759 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:46693: context canceled
I190719 09:11:59.798443 673760 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:43109: context canceled
I190719 09:11:59.798552 673761 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:44965: context canceled
--- FAIL: TestLargeUnsplittableRangeReplicate (66.54s)
    soon.go:35: condition failed to evaluate within 45s: not up-replicated yet. replicas: {1,2,3}
        goroutine 661549 [running]:
        runtime/debug.Stack(0x55dcc80, 0xc00eb9ab00, 0xc0084947f0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x55dcc80, 0xc00eb9ab00, 0xc0084947f0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/storage_test.TestLargeUnsplittableRangeReplicate(0xc00eb9ab00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replicate_queue_test.go:390 +0xbc3
        testing.tRunner(0xc00eb9ab00, 0x481c278)
        	/usr/local/go/src/testing/testing.go:865 +0x164
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:916 +0x65b

```

SHA: https://github.com/cockroachdb/cockroach/commits/22d48caaa7d39efdcef7b3c87a99fc421e1473af

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLargeUnsplittableRangeReplicate PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1397798&tab=buildLog

```
3      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190720 07:42:34.857815 698429 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:39711: context canceled
I190720 07:42:34.858567 698352 util/stop/stopper.go:542  quiescing; tasks left:
1      storage.IntentResolver: processing intents
W190720 07:42:34.859169 688788 storage/raft_transport.go:578  [n4] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
I190720 07:42:34.860156 698353 util/stop/stopper.go:542  quiescing; tasks left:
3      [async] closedts-subscription
I190720 07:42:34.860794 698353 util/stop/stopper.go:542  quiescing; tasks left:
2      [async] closedts-subscription
I190720 07:42:34.861358 698353 util/stop/stopper.go:542  quiescing; tasks left:
1      [async] closedts-subscription
I190720 07:42:34.863336 698354 util/stop/stopper.go:542  quiescing; tasks left:
3      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190720 07:42:34.863763 698354 util/stop/stopper.go:542  quiescing; tasks left:
3      [async] closedts-subscription
I190720 07:42:34.864225 698354 util/stop/stopper.go:542  quiescing; tasks left:
1      [async] closedts-subscription
I190720 07:42:34.883687 698455 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:39711: context canceled
I190720 07:42:34.886441 698456 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:36837: context canceled
I190720 07:42:34.886814 698457 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:46661: context canceled
I190720 07:42:34.886961 698458 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:34569: context canceled
I190720 07:42:34.887362 687971 storage/queue.go:518  [n3,s3] rate limited in MaybeAdd (gc): node unavailable; try another peer
I190720 07:42:34.887806 687971 storage/queue.go:518  [n3,s3] rate limited in MaybeAdd (merge): node unavailable; try another peer
I190720 07:42:34.888004 687971 storage/queue.go:518  [n3,s3] rate limited in MaybeAdd (split): node unavailable; try another peer
I190720 07:42:34.888151 687971 storage/queue.go:518  [n3,s3] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I190720 07:42:34.888256 687971 storage/queue.go:518  [n3,s3] rate limited in MaybeAdd (replicaGC): node unavailable; try another peer
I190720 07:42:34.888359 687971 storage/queue.go:518  [n3,s3] rate limited in MaybeAdd (raftlog): node unavailable; try another peer
I190720 07:42:34.888555 687971 storage/queue.go:518  [n3,s3] rate limited in MaybeAdd (raftsnapshot): node unavailable; try another peer
I190720 07:42:34.888618 687971 storage/queue.go:518  [n3,s3] rate limited in MaybeAdd (consistencyChecker): node unavailable; try another peer
I190720 07:42:34.888741 687971 storage/queue.go:518  [n3,s3] rate limited in MaybeAdd (timeSeriesMaintenance): node unavailable; try another peer
--- FAIL: TestLargeUnsplittableRangeReplicate (66.31s)
    soon.go:35: condition failed to evaluate within 45s: not up-replicated yet. replicas: {1,2,3}
        goroutine 686601 [running]:
        runtime/debug.Stack(0x5604c00, 0xc019a79f00, 0xc00b436750)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x5604c00, 0xc019a79f00, 0xc00b436750)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/storage_test.TestLargeUnsplittableRangeReplicate(0xc019a79f00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replicate_queue_test.go:390 +0xbc3
        testing.tRunner(0xc019a79f00, 0x4838cc8)
        	/usr/local/go/src/testing/testing.go:865 +0x164
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:916 +0x65b

```

SHA: https://github.com/cockroachdb/cockroach/commits/7dab0dcfd37c389af357c302c073b9611b5ada25

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLargeUnsplittableRangeReplicate PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1398579&tab=buildLog

```
I190721 07:31:10.589564 726064 util/stop/stopper.go:542  quiescing; tasks left:
1      [async] transport racer
I190721 07:31:10.556663 726067 util/stop/stopper.go:542  quiescing; tasks left:
4      [async] closedts-subscription
1      [async] intent_resolver_ir_batcher
1      [async] closedts-rangefeed-subscriber
I190721 07:31:10.590554 726067 util/stop/stopper.go:542  quiescing; tasks left:
4      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190721 07:31:10.591952 726067 util/stop/stopper.go:542  quiescing; tasks left:
3      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190721 07:31:10.592426 726067 util/stop/stopper.go:542  quiescing; tasks left:
2      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190721 07:31:10.592911 726067 util/stop/stopper.go:542  quiescing; tasks left:
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190721 07:31:10.611074 726213 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:43883: context canceled
I190721 07:31:10.611291 726215 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:45449: context canceled
I190721 07:31:10.612873 726216 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:41353: context canceled
I190721 07:31:10.613967 726214 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:41981: context canceled
I190721 07:31:10.636028 726126 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:33823: context canceled
I190721 07:31:10.637013 726127 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:43883: context canceled
I190721 07:31:10.637575 726129 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:41353: context canceled
I190721 07:31:10.637921 726128 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:45449: context canceled
I190721 07:31:10.680025 726220 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:43883: context canceled
I190721 07:31:10.681342 726219 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:33823: context canceled
I190721 07:31:10.682508 726221 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:45449: context canceled
I190721 07:31:10.683686 726222 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:41981: context canceled
I190721 07:31:10.702762 726176 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:41981: context canceled
I190721 07:31:10.785028 714312 kv/transport_race.go:108  transport race promotion: ran 65 iterations on up to 4273 requests
--- FAIL: TestLargeUnsplittableRangeReplicate (68.65s)
    soon.go:35: condition failed to evaluate within 45s: not up-replicated yet. replicas: {1,2,3}
        goroutine 713970 [running]:
        runtime/debug.Stack(0x5606580, 0xc008e59b00, 0xc014c09180)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x5606580, 0xc008e59b00, 0xc014c09180)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/storage_test.TestLargeUnsplittableRangeReplicate(0xc008e59b00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replicate_queue_test.go:390 +0xbc3
        testing.tRunner(0xc008e59b00, 0x483a180)
        	/usr/local/go/src/testing/testing.go:865 +0x164
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:916 +0x65b

```

SHA: https://github.com/cockroachdb/cockroach/commits/1ad0ecc8cbddf82c9fedb5a5c5e533e72a657ff7

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLargeUnsplittableRangeReplicate PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1399377&tab=buildLog

```
1      [async] closedts-rangefeed-subscriber
W190722 10:40:49.498161 649577 storage/raft_transport.go:578  [n5] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
I190722 10:40:49.499724 648676 rpc/nodedialer/nodedialer.go:139  [n4] unable to connect to n2: failed to connect to n2 at 127.0.0.1:42897: context canceled
I190722 10:40:49.500633 658450 util/stop/stopper.go:542  quiescing; tasks left:
2      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190722 10:40:49.502644 658450 util/stop/stopper.go:542  quiescing; tasks left:
2      [async] closedts-subscription
I190722 10:40:49.503111 658450 util/stop/stopper.go:542  quiescing; tasks left:
1      [async] closedts-subscription
I190722 10:40:49.503336 648676 rpc/nodedialer/nodedialer.go:139  [n4] unable to connect to n3: failed to connect to n3 at 127.0.0.1:38785: context canceled
W190722 10:40:49.504064 648965 storage/raft_transport.go:578  [n4] while processing outgoing Raft queue to node 1: EOF:
W190722 10:40:49.504762 651305 storage/raft_transport.go:578  [n4] while processing outgoing Raft queue to node 5: rpc error: code = Canceled desc = grpc: the client connection is closing:
I190722 10:40:49.506634 658385 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:40907: context canceled
I190722 10:40:49.506968 658383 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:46067: context canceled
I190722 10:40:49.505131 658382 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:34629: context canceled
I190722 10:40:49.505583 658452 util/stop/stopper.go:542  quiescing; tasks left:
2      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190722 10:40:49.506222 658384 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:38785: context canceled
I190722 10:40:49.506792 658449 util/stop/stopper.go:542  quiescing; tasks left:
1      [async] closedts-subscription
I190722 10:40:49.506907 658501 util/stop/stopper.go:542  quiescing; tasks left:
2      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190722 10:40:49.507828 658452 util/stop/stopper.go:542  quiescing; tasks left:
2      [async] closedts-subscription
I190722 10:40:49.508839 658501 util/stop/stopper.go:542  quiescing; tasks left:
2      [async] closedts-subscription
I190722 10:40:49.521000 658318 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:46067: context canceled
I190722 10:40:49.522345 658316 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:40907: context canceled
I190722 10:40:49.522890 658319 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:42897: context canceled
I190722 10:40:49.524467 658317 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:38785: context canceled
--- FAIL: TestLargeUnsplittableRangeReplicate (64.48s)
    soon.go:35: condition failed to evaluate within 45s: not up-replicated yet. replicas: {1,2,3}
        goroutine 647008 [running]:
        runtime/debug.Stack(0x5606580, 0xc0152ee000, 0xc001242ae0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x5606580, 0xc0152ee000, 0xc001242ae0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/storage_test.TestLargeUnsplittableRangeReplicate(0xc0152ee000)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replicate_queue_test.go:390 +0xbc3
        testing.tRunner(0xc0152ee000, 0x483a180)
        	/usr/local/go/src/testing/testing.go:865 +0x164
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:916 +0x65b

```

SHA: https://github.com/cockroachdb/cockroach/commits/7111a67b2ea3a19c2f312f8d214b8823f431cac0

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLargeUnsplittableRangeReplicate PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1401330&tab=buildLog

```
W190723 10:34:29.174451 653464 storage/raft_transport.go:578  [n3] while processing outgoing Raft queue to node 4: rpc error: code = Canceled desc = grpc: the client connection is closing:
I190723 10:34:29.175166 663348 util/stop/stopper.go:542  quiescing; tasks left:
2      [async] closedts-subscription
I190723 10:34:29.175387 663351 util/stop/stopper.go:542  quiescing; tasks left:
4      [async] closedts-subscription
W190723 10:34:29.175832 652750 storage/raft_transport.go:578  [n2] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W190723 10:34:29.176925 659414 storage/raft_transport.go:578  [n4] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W190723 10:34:29.177791 653367 gossip/gossip.go:1498  [n5] no incoming or outgoing connections
I190723 10:34:29.176058 663351 util/stop/stopper.go:542  quiescing; tasks left:
3      [async] closedts-subscription
W190723 10:34:29.176555 655664 storage/raft_transport.go:578  [n5] while processing outgoing Raft queue to node 4: rpc error: code = Unavailable desc = transport is closing:
W190723 10:34:29.179941 653901 storage/raft_transport.go:578  [n5] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
I190723 10:34:29.182774 663352 util/stop/stopper.go:542  quiescing; tasks left:
3      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190723 10:34:29.183545 663352 util/stop/stopper.go:542  quiescing; tasks left:
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190723 10:34:29.183940 663352 util/stop/stopper.go:542  quiescing; tasks left:
1      [async] closedts-rangefeed-subscriber
I190723 10:34:29.186210 663335 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:37179: context canceled
I190723 10:34:29.187512 663334 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:38541: context canceled
I190723 10:34:29.192628 663337 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:40943: context canceled
I190723 10:34:29.193056 663355 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:40943: context canceled
I190723 10:34:29.194529 663353 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:41091: context canceled
I190723 10:34:29.195022 663336 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:41429: context canceled
I190723 10:34:29.196481 663354 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:41429: context canceled
I190723 10:34:29.197626 663356 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:37179: context canceled
--- FAIL: TestLargeUnsplittableRangeReplicate (67.73s)
    soon.go:35: condition failed to evaluate within 45s: not up-replicated yet. replicas: {1,2,3}
        goroutine 651330 [running]:
        runtime/debug.Stack(0x5648b20, 0xc00975f100, 0xc00a799390)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x5648b20, 0xc00975f100, 0xc00a799390)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/storage_test.TestLargeUnsplittableRangeReplicate(0xc00975f100)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replicate_queue_test.go:390 +0xbc3
        testing.tRunner(0xc00975f100, 0x487d450)
        	/usr/local/go/src/testing/testing.go:865 +0x164
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:916 +0x65b

```

SHA: https://github.com/cockroachdb/cockroach/commits/86eab2ff0a1a4c2d9b5f7e7a45deda74c98c6c37

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLargeUnsplittableRangeReplicate PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1403292&tab=buildLog

```
I190724 08:14:18.313815 689265 rpc/nodedialer/nodedialer.go:139  [n2] unable to connect to n5: failed to connect to n5 at 127.0.0.1:44693: context canceled
I190724 08:14:18.322421 690634 rpc/nodedialer/nodedialer.go:139  [n5] unable to connect to n4: failed to connect to n4 at 127.0.0.1:43771: context canceled
I190724 08:14:18.323115 690634 rpc/nodedialer/nodedialer.go:139  [n5] unable to connect to n3: failed to connect to n3 at 127.0.0.1:39183: context canceled
I190724 08:14:18.383469 700215 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:44693: context canceled
I190724 08:14:18.383738 700216 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:42215: context canceled
I190724 08:14:18.384012 700217 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:36637: context canceled
E190724 08:14:18.384226 700219 vendor/google.golang.org/grpc/pickfirst.go:59  pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
I190724 08:14:18.384468 700218 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:43771: context canceled
I190724 08:14:18.397265 700190 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:43771: context canceled
I190724 08:14:18.397646 700189 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:39183: context canceled
I190724 08:14:18.415349 700159 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:39183: context canceled
I190724 08:14:18.415665 700162 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:42215: context canceled
I190724 08:14:18.420561 700247 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:42215: context canceled
I190724 08:14:18.421485 700245 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:36637: context canceled
I190724 08:14:18.429850 688433 kv/transport_race.go:108  transport race promotion: ran 149 iterations on up to 4111 requests
W190724 08:14:18.443236 690636 storage/store.go:1537  [n5,s5,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I190724 08:14:18.459866 700132 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n5: failed to connect to n5 at 127.0.0.1:44693: context canceled
I190724 08:14:18.461354 700130 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:36637: context canceled
I190724 08:14:18.463510 700261 rpc/nodedialer/nodedialer.go:139  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:39183: context canceled
W190724 08:14:18.489307 690636 storage/store.go:1537  [n5,s5,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
--- FAIL: TestLargeUnsplittableRangeReplicate (65.90s)
    soon.go:35: condition failed to evaluate within 45s: not up-replicated yet. replicas: {1,2,3}
        goroutine 688244 [running]:
        runtime/debug.Stack(0x5657480, 0xc017270f00, 0xc005bbbb50)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x5657480, 0xc017270f00, 0xc005bbbb50)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/storage_test.TestLargeUnsplittableRangeReplicate(0xc017270f00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replicate_queue_test.go:390 +0xbc3
        testing.tRunner(0xc017270f00, 0x488a940)
        	/usr/local/go/src/testing/testing.go:865 +0x164
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:916 +0x65b

```

--- a/pkg/storage/replicate_queue_test.go
+++ b/pkg/storage/replicate_queue_test.go
@@ -378,16 +378,19 @@ func TestLargeUnsplittableRangeReplicate(t *testing.T) {
 	_, err = db.Exec("alter table t configure zone using num_replicas = 5")
 	require.NoError(t, err)
 
-	// Speed up the queue processing.
-	for _, s := range tc.Servers {
-		err := s.Stores().VisitStores(func(store *storage.Store) error {
-			return store.ForceReplicationScanAndProcess()
-		})
-		require.NoError(t, err)
+	forceProcess := func() {
+		// Speed up the queue processing.
+		for _, s := range tc.Servers {
+			err := s.Stores().VisitStores(func(store *storage.Store) error {
+				return store.ForceReplicationScanAndProcess()
+			})
+			require.NoError(t, err)
+		}
 	}
 
 	// Wait until the smaller range (the 2nd) has up-replicated.
 	testutils.SucceedsSoon(t, func() error {
+		forceProcess()
 		r := db.QueryRow(
 			"select replicas from [show experimental_ranges from table t] where start_key='/2'")
 		var repl string
@@ -402,6 +405,7 @@ func TestLargeUnsplittableRangeReplicate(t *testing.T) {
 
 	// Now check that the large range also gets up-replicated.
 	testutils.SucceedsSoon(t, func() error {
+		forceProcess()
 		r := db.QueryRow(
 			"select replicas from [show experimental_ranges from table t] where start_key is null")
 		var repl string

